src/components/LmsConfigurations/SuccessFactorsConfigForm.jsx

```diff
@@ -0,0 +1,374 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import NewRelicService from '../../data/services/NewRelicService';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+
+export const REQUIRED_SUCCESS_FACTOR_CONFIG_FIELDS = [
+  'sapsfBaseUrl',
+  'sapsfCompanyId',
+  'sapsfUserId',
+  'key',
+  'secret',
+  'userType',
+];
+
+class SuccessFactorsIntegrationConfigForm extends React.Component {
```

Comment:
Don't bother with this one because it's not worth your time to refactor, but we're moving away from class components on the frontend in favor of function components, so future work should be functional React components. Happy to chat about what that looks like next time you pick up a frontend ticket!

Reply:
yeah I've done purely functional components before. This style was also already more functional to me in comparison to previous class component work I've done at other places (there is no constructor at all here) so I wasn't sure if this constructor-less class component was the preferred style. Guess not!

Reply:
Yeah, no worries, no way you could have known; just wanted to pass info along for the future.
Personally, I think that even though React says they'll never get rid of class components, they are eventually going to ditch support for class components.

Reply:
What do you think about me throwing in a ticket to refactor this repo to use functional components?
I think it's totally fine to want to move to something we decide is better stylistically/programmatically, I just also want us to standardize. From a glance it's very hard to gather what "way" we want our frontend components done since they all end up different lol.

I'd love to tackle this all at once.

Reply:
Honestly, I don't think refactoring to functional components is *ever* worth the time it takes. For new components, I think we should write them as functional components. But it's a different enough way of thinking about things that in general I don't think it's worth our time until React's inevitable betrayal.
My take is, if the class component works, use it.
The refactor I *do* want to open a ticket for is a refactor to share the error handling and other code that is re-used between this and Alex's PR. And that's a good opportunity to move the form part of this to a functional component.